debuginfo/src/dwarf.rs

```diff
@@ -635,6 +661,16 @@ impl<'d, 'a> DwarfUnit<'d, 'a> {
             range_buf.clear();
             let (call_line, call_file) = self.parse_ranges(entry, range_buf)?;
 
+            // TODO: fix that
+            // We've a non-inlined function which has two ranges or more
+            // So probably splited because of cold paths
+            // Anyway the function_size & function_end are wrong here
+            // so probably the workaround to generate several functions here:
+            // one for each range
+            if !inline && range_buf.len() != 1 {
+                continue;
```

Comment:
Can you join this with the next check for an empty `range_buf`, or at least also set `skipped_depth`? This way, you can remove the second part of this TODO, where you skip in the inline child.

I've quickly run a few tests, and it seems that this doesn't produce any false positives on the files that I've checked it on. However, maybe we want to check for explicit _split ranges_? At least in inline functions, it's quite common to get adjacent ranges, and I'm not sure if we can trust that this doesn't happen for top-level functions as well.